source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRSeparator.mm

```diff
@@ -0,0 +1,170 @@
+                separator.lineWidth = separatorHstCnfig->lineThickness;
+                separator.backgroundColor = UIColor.clearColor;
+                [superView addArrangedSubview:separator];
+                
+                NSLayoutConstraint* constraint = nil;
+                separator.axis = superView.axis;
+                if(UILayoutConstraintAxisVertical == superView.axis)
+                {
+                    constraint = [NSLayoutConstraint constraintWithItem: separator
+                                                              attribute: NSLayoutAttributeWidth
+                                                              relatedBy: NSLayoutRelationEqual
+                                                                 toItem: superView
+                                                              attribute: NSLayoutAttributeWidth
+                                                             multiplier: 1
+                                                               constant: 0];
+                    [separator setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
+                    [separator setContentCompressionResistancePriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
+                    [separator setContentHuggingPriority:UILayoutPriorityDefaultHigh forAxis:UILayoutConstraintAxisVertical];
+                    [separator setContentCompressionResistancePriority:UILayoutPriorityDefaultHigh forAxis:UILayoutConstraintAxisVertical];
+                }
+                else
+                {
+                    constraint = [NSLayoutConstraint constraintWithItem: separator
+                                                              attribute: NSLayoutAttributeHeight
+                                                              relatedBy: NSLayoutRelationEqual
+                                                                 toItem: superView
+                                                              attribute: NSLayoutAttributeHeight
+                                                             multiplier: 1
+                                                               constant: 0];
+                    [separator setContentHuggingPriority:UILayoutPriorityDefaultHigh forAxis:UILayoutConstraintAxisHorizontal];
+                    [separator setContentCompressionResistancePriority:UILayoutPriorityDefaultHigh forAxis:UILayoutConstraintAxisHorizontal];
+                    [separator setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisVertical];
+                    [separator setContentCompressionResistancePriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisVertical];
+                }
+                
+                if(constraint) [superView addConstraint:constraint];
+            }
+            catch(...) {;};
```

Comment:
intended not to created seperator if wrongly formatted. Thanks for your insight, and changed as suggested.


---
In reply to: [125698488](https://github.com/Microsoft/AdaptiveCards/pull/454#discussion_r125698488) [](ancestors = 125698488)